code/game/objects/items/devices/enginepicker.dm

```diff
@@ -0,0 +1,80 @@
+/*
+//////////
+	Item meant to spawn one of the three (Tesla / Singularity / Supermatter) engines on-station at round-start.
+	Should be found in the CE's office. Not access-restricted.
+//////////
+*/
+
+/obj/item/enginepicker
+	name = "Bluespace Engine Delivery Device"
+	desc = "With all of the innovations in power-generating science, Nanotrasen has decided to invent a per-station bluespace-based delivery system for a unique engine of the Chief Engineer's choice. Only one choice can be made. Device self-destructs on use."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "enginepicker"
+
+	var/list/list_enginebeacons = list()
+
+/obj/item/enginepicker/attack_self(mob/living/carbon/user)
+	if(usr.stat || !usr.canmove || usr.restrained())
+		return
+	locatebeacons()
+	var/default = null
+	var/E = input("Select the station's Engine:", "[src]", default) as null|anything in list_enginebeacons
+	if(E)
+		processchoice(E, user)
+
+//This proc finds all of the '/obj/item/radio/beacon/engine' in the world and assigns them to a list
+/obj/item/enginepicker/proc/locatebeacons()
+	LAZYCLEARLIST(list_enginebeacons)
+	for(var/obj/item/radio/beacon/engine/B in world)
```

Comment:
`world` searches are awful. Make a global list and add the beacons to the list on `Initialize()`, similar to what I did in #10953

Reply:
Done!